File: VB/CustomTableOfContents/CategoriesReport.vb
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace CustomTableOfContents

    Public Partial Class CategoriesReport
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub lbCategoryName_BeforePrint(ByVal sender As Object, ByVal e As CancelEventArgs)
            Dim label As XRControl = TryCast(sender, XRControl)
            label.Tag = String.Format("Link_{0}", GetCurrentColumnValue("CategoryID"))
        End Sub

        Private Sub lbPage_BeforePrint(ByVal sender As Object, ByVal e As CancelEventArgs)
            Dim label As XRControl = TryCast(sender, XRControl)
            label.Tag = String.Format("PageBrick_{0}", GetCurrentColumnValue("CategoryID"))
        End Sub
    End Class
End Namespace

File: CS/CustomTableOfContents/CategoriesReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace CustomTableOfContents {
    public partial class CategoriesReport : DevExpress.XtraReports.UI.XtraReport {
        public CategoriesReport() {
            InitializeComponent();
        }

        private void lbCategoryName_BeforePrint(object sender, CancelEventArgs e) {
            XRControl label = sender as XRControl;
            label.Tag = String.Format("Link_{0}", GetCurrentColumnValue("CategoryID"));
        }

        private void lbPage_BeforePrint(object sender, CancelEventArgs e) {
            XRControl label = sender as XRControl;
            label.Tag = String.Format("PageBrick_{0}", GetCurrentColumnValue("CategoryID"));
        }

    }
}
